com.hazelcast.internal.util.phonehome.PhoneHomeIntegrationTest.testForCloudIfGCP [HZ-1156]

_5.0.z_ (commit 494623f0b4f71742b7e9c006c8dc2af0f26c4b57)

Failed on Corretto JDK16: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-5.maintenance-CorrettoJDK16/11/testReport/com.hazelcast.internal.util.phonehome/PhoneHomeIntegrationTest/testForCloudIfGCP/

<details><summary>Stacktrace:</summary>

```
com.github.tomakehurst.wiremock.common.FatalStartupException: java.lang.RuntimeException: java.io.IOException: Failed to bind to /0.0.0.0:8080
	at com.github.tomakehurst.wiremock.WireMockServer.start(WireMockServer.java:148)
	at com.github.tomakehurst.wiremock.junit.WireMockRule$1.evaluate(WireMockRule.java:69)
	at com.hazelcast.test.DumpBuildInfoOnFailureRule$1.evaluate(DumpBuildInfoOnFailureRule.java:37)
	at com.hazelcast.test.jitter.JitterRule$1.evaluate(JitterRule.java:104)
	at com.hazelcast.test.metrics.MetricsRule$1.evaluate(MetricsRule.java:62)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:50)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:29)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at com.hazelcast.test.AfterClassesStatement.evaluate(AfterClassesStatement.java:39)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: java.lang.RuntimeException: java.io.IOException: Failed to bind to /0.0.0.0:8080
	at com.github.tomakehurst.wiremock.jetty9.JettyHttpServer.start(JettyHttpServer.java:191)
	at com.github.tomakehurst.wiremock.WireMockServer.start(WireMockServer.java:146)
	... 37 more
Caused by: java.io.IOException: Failed to bind to /0.0.0.0:8080
	at wiremock.org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:346)
	at wiremock.org.eclipse.jetty.server.ServerConnector.open(ServerConnector.java:307)
	at wiremock.org.eclipse.jetty.server.AbstractNetworkConnector.doStart(AbstractNetworkConnector.java:80)
	at wiremock.org.eclipse.jetty.server.ServerConnector.doStart(ServerConnector.java:231)
	at wiremock.org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:72)
	at wiremock.org.eclipse.jetty.server.Server.doStart(Server.java:385)
	at wiremock.org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:72)
	at com.github.tomakehurst.wiremock.jetty9.JettyHttpServer.start(JettyHttpServer.java:189)
	... 38 more
Caused by: java.net.BindException: Address already in use
	at java.base/sun.nio.ch.Net.bind0(Native Method)
	at java.base/sun.nio.ch.Net.bind(Net.java:552)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.netBind(ServerSocketChannelImpl.java:336)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:294)
	at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:88)
	at wiremock.org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:342)
	... 45 more
```

</details>

<details><summary>Standard output:</summary>

```
Started Running Test: testForCloudIfGCP
BuildInfo right after testForCloudIfGCP(com.hazelcast.internal.util.phonehome.PhoneHomeIntegrationTest): BuildInfo{version='5.0.2-SNAPSHOT', build='20211205', buildNumber=20211205, revision=494623f, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'testForCloudIfGCP(com.hazelcast.internal.util.phonehome.PhoneHomeIntegrationTest):'
02:32:25, accumulated pauses: 34 ms, max pause: 4 ms, pauses over 1000 ms: 0```

</details>

--- a/hazelcast/src/test/java/com/hazelcast/internal/util/phonehome/PhoneHomeIntegrationTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/util/phonehome/PhoneHomeIntegrationTest.java
@@ -59,6 +59,7 @@ import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
 import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
 import static com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo;
 import static com.github.tomakehurst.wiremock.client.WireMock.verify;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
 import static com.hazelcast.cache.CacheTestSupport.createServerCachingProvider;
 import static com.hazelcast.test.Accessors.getNode;
 import static org.mockito.Mockito.when;
@@ -72,7 +73,7 @@ public class PhoneHomeIntegrationTest extends HazelcastTestSupport {
     }
 
     @Rule
-    public WireMockRule wireMockRule = new WireMockRule();
+    public WireMockRule wireMockRule = new WireMockRule(wireMockConfig().dynamicPort());
 
     private Node node;
     private PhoneHome phoneHome;
@@ -94,11 +95,12 @@ public class PhoneHomeIntegrationTest extends HazelcastTestSupport {
         when(dockerPath.toRealPath()).thenReturn(Paths.get(System.getProperty("user.dir")));
         when(kubernetesTokenPath.toRealPath()).thenReturn(Paths.get(System.getProperty("user.dir")));
 
-        cloudInfoCollector = new CloudInfoCollector("http://localhost:8080/latest/meta-data",
-                "http://localhost:8080/metadata/instance/compute?api-version=2018-02-01",
-                "http://localhost:8080/metadata.google.internal", kubernetesTokenPath, dockerPath);
+        int port = wireMockRule.port();
+        cloudInfoCollector = new CloudInfoCollector("http://localhost:" + port + "/latest/meta-data",
+                "http://localhost:" + port + "/metadata/instance/compute?api-version=2018-02-01",
+                "http://localhost:" + port + "/metadata.google.internal", kubernetesTokenPath, dockerPath);
 
-        phoneHome = new PhoneHome(node, "http://localhost:8080/ping", cloudInfoCollector);
+        phoneHome = new PhoneHome(node, "http://localhost:" + port + "/ping", cloudInfoCollector);
         stubUrls("200", "4XX", "4XX", "4XX");
     }
 

--- a/hazelcast/src/test/java/com/hazelcast/internal/util/phonehome/RESTClientPhoneHomeTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/util/phonehome/RESTClientPhoneHomeTest.java
@@ -41,6 +41,7 @@ import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
 import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
 import static com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo;
 import static com.github.tomakehurst.wiremock.client.WireMock.verify;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
 import static com.hazelcast.internal.ascii.HTTPCommunicator.URI_MAPS;
 import static com.hazelcast.internal.ascii.HTTPCommunicator.URI_QUEUES;
 import static com.hazelcast.internal.util.phonehome.PhoneHomeIntegrationTest.containingParam;
@@ -60,13 +61,14 @@ public class RESTClientPhoneHomeTest {
     }
 
     @Rule
-    public WireMockRule wireMockRule = new WireMockRule();
+    public WireMockRule wireMockRule = new WireMockRule(wireMockConfig().dynamicPort());
 
     private HazelcastInstance instance;
 
     private HTTPCommunicator http;
 
     private Config config;
+    private int port;
 
     @Before
     public void setUp() {
@@ -74,6 +76,7 @@ public class RESTClientPhoneHomeTest {
         instance = factory.newHazelcastInstance(config);
         http = new HTTPCommunicator(instance);
         stubFor(post(urlPathEqualTo("/ping")).willReturn(aResponse().withStatus(200)));
+        port = wireMockRule.port();
     }
 
     @After
@@ -103,7 +106,7 @@ public class RESTClientPhoneHomeTest {
         assertEquals(400, http.doGet(http.getUrl(URI_MAPS)).responseCode);
         assertEquals(200, http.mapDelete("my-map", "key"));
 
-        PhoneHome phoneHome = new PhoneHome(getNode(instance), "http://localhost:8080/ping");
+        PhoneHome phoneHome = new PhoneHome(getNode(instance), "http://localhost:" + port + "/ping");
         phoneHome.phoneHome(false);
 
         verify(1, postRequestedFor(urlPathEqualTo("/ping"))
@@ -134,7 +137,7 @@ public class RESTClientPhoneHomeTest {
         assertEquals(204, http.doDelete(http.getUrl(URI_QUEUES) + "my-queue/10").responseCode);
         assertEquals(400, http.doDelete(http.getUrl(URI_QUEUES) + "my-queue").responseCode);
 
-        PhoneHome phoneHome = new PhoneHome(getNode(instance), "http://localhost:8080/ping");
+        PhoneHome phoneHome = new PhoneHome(getNode(instance), "http://localhost:" + port + "/ping");
         phoneHome.phoneHome(false);
 
         verify(1, postRequestedFor(urlPathEqualTo("/ping"))
@@ -157,7 +160,7 @@ public class RESTClientPhoneHomeTest {
         http.configReload(config.getClusterName(), "");
         http.configUpdate(config.getClusterName(), "", "hazelcast:\n");
 
-        PhoneHome phoneHome = new PhoneHome(getNode(instance), "http://localhost:8080/ping");
+        PhoneHome phoneHome = new PhoneHome(getNode(instance), "http://localhost:" + port + "/ping");
         phoneHome.phoneHome(false);
 
         verify(1, postRequestedFor(urlPathEqualTo("/ping"))

